app/assets/javascript/app/containers/ViewerContainer.jsx

```diff
@@ -3,13 +3,17 @@ import Viewer from '../components/Viewer';
 
 import { fetchRoads } from '../actions/roadActions';
 import { fetchCity } from '../actions/cityActions';
+import { logout } from '../actions/sessionActions';
 import { fetchPlans } from '../actions/planActions';
 import { fetchNodes } from '../actions/nodeActions';
 
 const mapStateToProps = (state, props) => state;
 
 const mapDispatchToProps = (dispatch, props) => {
   return {
+    fetchRoads: () => dispatch(fetchRoads(props.match.params.city)),
+    fetchCity: () => dispatch(fetchCity(props.match.params.city)),
```

Comment:
We don't need to fetchCity and fetchRoads here anymore. They're fetched based on the route.